signaler/signaler_test.go

```diff
@@ -0,0 +1,37 @@
+//go:build !windows
+// +build !windows
+
+package signaler
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWaitForInterrupt_SIGTERM(t *testing.T) {
```

Comment:
🚧 Unusual naming... why would we test just one, and why put it in the test name?
```suggestion
func TestWaitForInterrupt(t *testing.T) {
```